[4657] Reactify Global App (Audit)

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4657

aachavarria committed:
Creating AuditManagement.tsx;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/audit-reactify

aachavarria committed:
Adding Basic Fetch and AuditManagement Files;

aachavarria committed:
exporting interface;

rart commented on ui/app/src/services/audit.ts:
```diff
@@ -22,13 +22,15 @@ import { PagedArray } from '../models/PagedArray';
 import { AuditLog } from '../models/Audit';
 
 export function fetchAudit(options): Observable<PagedArray<AuditLog>> {
```
Please type `options`

aachavarria committed:
Creating DataGrid on AuditGridUI.tsx;
Updating AuditManagement.tsx;

aachavarria committed:
adding fetchSpecificAudit();

rart commented on ui/app/src/services/audit.ts:
```diff
@@ -20,22 +20,42 @@ import { map, pluck } from 'rxjs/operators';
+  }
+>;
+
+export function fetchAudit(options: AuditOptions): Observable<PagedArray<AuditLog>> {
```
fetchAudit => fetchAuditLog
AuditLog => AuditLogEntry

rart commented on ui/app/src/services/audit.ts:
```diff
@@ -20,22 +20,42 @@ import { map, pluck } from 'rxjs/operators';
 }
 
-export function fetchSpecificAudit(id: number): Observable<AuditLog[]> {
+export function fetchSpecificAudit(id: number): Observable<AuditLog> {
```
fetchSpecificAudit => fetchAuditLogEntry

aachavarria committed:
Fixing Legacy Audit;
AuditGridUI: Adding TimeStamp Filter;
AuditGridUI: Formatting Date;
Creating ParametersDialog.tsx;
PR Feedback;

aachavarria committed:
Adding resetFilters Support;
WIP: ActiveFilters Icon/State, Legacy Implementation, fromTo Date Filtering

aachavarria committed:
adding date filtering;
TODO: Date validation, Add timeSelector;

aachavarria committed:
Refactor creating AuditGridFilterPopover.tsx

aachavarria committed:
Adding Clear Buttons

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/global-menu/audit-reactify

aachavarria committed:
Legacy Integration

aachavarria committed:
Creating AuditGridSkeleton.tsx

aachavarria committed:
PR Feedback

aachavarria committed:
adding siteMode prop;

aachavarria committed:
adding autoHeight

rart commented on ui/app/src/components/AuditGrid/AuditGridSkeleton.tsx:
```diff
@@ -0,0 +1,199 @@
+
+import Box from '@material-ui/core/Box';
+import React, { useMemo } from 'react';
+import { styles } from './styles';
```
rename styles => useStyles or use{componentName}Styles if there are multiple

rart commented on ui/app/src/components/ParametersDialog/index.ts:
```diff
@@ -0,0 +1,19 @@
+/*
```
Rename as AuditLogEntryParametersDialog

rart replied:
Missing changing the folder name to `AuditLogEntryParametersDialog`

rart commented on ui/app/src/models/Audit.ts:
```diff
@@ -14,14 +14,14 @@
  */
 
-interface LogParameters {
+export interface LogParameters {
```
LogParameters => AuditLogEntryParameter

rart commented on ui/app/src/components/ParametersDialog/ParametersDialog.tsx:
```diff
@@ -0,0 +1,91 @@
+  parameters: LogParameters[];
+}
+
+export default function ParametersDialog(props: ParametersDialogProps) {
```
AuditLogEntryParametersDialog

rart commented on ui/app/src/components/AuditGridFilterPopover/styles.ts:
```diff
@@ -0,0 +1,48 @@
+
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+
+export const styles = makeStyles((theme) =>
```
=> useStyles

rart commented on ui/app/src/components/AuditGrid/styles.ts:
```diff
@@ -0,0 +1,56 @@
+
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+
+export const styles = makeStyles((theme) =>
```
=> useStyles

rart commented on ui/app/src/components/AuditGridFilterPopover/AuditGridFilterPopover.tsx:
```diff
@@ -0,0 +1,341 @@
+  );
+}
+
+export function AuditGridFilterPopoverContainer(props: AuditGridFilterPopoverProps) {
```
- => AuditGridFilterPopoverBody
- Move to own file

aachavarria committed:
PR Feedback

aachavarria committed:
Renaming

rart commented on ui/app/src/components/AuditLogEntryParametersDialog/AuditLogEntryParametersDialogUI.tsx:
```diff
@@ -0,0 +1,74 @@
+    return () => {
+      onClosed?.();
+    };
+  });
```
useUnmount!
```suggestion
  useMount(onClosed);
```

aachavarria committed:
Using useUnmount
